### PR TITLE
Added taxonomy-insights-documentation.md

### PR DESCRIPTION
**Pull request by juancappi**
This is not a knowledge submission, it's user-facing documentation about PR insights workflow and reporting.

@jjasghar @bjhargrave @lhawthorn 
@renrenstart @nitin-ramchandani

### CODE REVIEW

**Comment by jjasghar** [comment]
Looks great, can you take care of the lint and DCO bot issues?

**Comment by bjhargrave** [comment]
It is not stated whether insights also analyzes knowledge documents. Can we state, in this document, how knowledge documents are analyzed or not?

File: docs/triaging/insights/images/InstructLab-Insights-report.png
[the remark this answers is not in the document]
Comment (juancappi) [reply docs/triaging/insights/images/InstructLab-Insights-report.png:1]: Thanks for pointing this out. Addressed

File: docs/triaging/insights/taxonomy-insights-documentation.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# InstructLab Contribution Insights
+
+This module automatically analyzes the taxonomy of pull request contribution files, checking for potential code of conduct violations.
```
Comment (bjhargrave) [inline docs/triaging/insights/taxonomy-insights-documentation.md:3]: It is not stated whether insights also analyzes knowledge documents. Can we state, in this document, how knowledge documents are analyzed or not?

**Review by bjhargrave — CHANGES_REQUESTED** [review]

File: docs/triaging/insights/taxonomy-insights-documentation.md (hunk 1 above)
Comment (juancappi) [reply docs/triaging/insights/taxonomy-insights-documentation.md:3]: Analysis for the referenced markdown files is out of scope for this initial version, but it's in plan. I've made an explicit mention. Let me know if the text is clear enough.

**Review by bjhargrave — APPROVED** [review]

**Review by danmcp — APPROVED** [review]

**Pull request merged**